lib/streamlit/runtime/state/query_params_proxy.py

```diff
@@ -205,3 +207,19 @@ def from_dict(self, params):
         """
         with get_session_state().query_params() as qp:
             return qp.from_dict(params)
+
+    @gather_metrics("query_params.to_string")
+    def to_string(self) -> str:
```

Comment:
super nit: If this is internal, should probably be called `_to_string`

Reply:
It's in both...? Because we don't know if we're going to receive a proxy or the underlying object.

Technically users can use this to generat easy "permalinks" for people to copy, so it's ok in my mind to have it be private.

Reply:
Ah oops sorry misread that sorry 😨